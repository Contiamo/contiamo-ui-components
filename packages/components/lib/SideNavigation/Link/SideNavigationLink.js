"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var utils_1 = require("@operational/utils");
var style = function (_a) {
    var theme = _a.theme, color = _a.color;
    var backgroundColor = color ? utils_1.hexOrColor(color)(theme.colors.palette[color]) : theme.colors.palette.grey90;
    return {
        backgroundColor: backgroundColor,
        position: "relative",
        zIndex: theme.baseZIndex + 1,
        margin: "0 " + theme.spacing * -0.5 + "px",
        padding: theme.spacing + "px",
        minWidth: 200,
        borderRadius: 2,
        transition: ".1s background-color ease",
        color: utils_1.readableTextColor(backgroundColor)(["black", "white"]),
        "& + &": {
            borderTop: "1px solid " + theme.colors.palette.grey90
        },
        ":hover": {
            backgroundColor: utils_1.darken(backgroundColor)(10)
        },
        ":focus": {
            outline: 0,
            backgroundColor: utils_1.darken(backgroundColor)(15)
        },
        ":first-child": {
            marginTop: theme.spacing * -0.5 + "px"
        },
        ":last-child": {
            marginBottom: theme.spacing * -0.5 + "px"
        }
    };
};
exports.style = style;
var SideNavigationLink = function (props) { return (React.createElement("div", { key: props.id, style: props.style, className: props.className, onClick: props.onClick, role: "button", tabIndex: -1 }, props.children)); };
exports.SideNavigationLink = SideNavigationLink;
exports.default = glamorous_1.default(SideNavigationLink)(style);
//# sourceMappingURL=SideNavigationLink.js.map