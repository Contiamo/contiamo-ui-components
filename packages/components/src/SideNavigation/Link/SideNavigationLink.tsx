import * as React from "react"
import glamorous, { GlamorousComponent } from "glamorous"
import { Theme } from "@operational/theme"
import { hexOrColor, readableTextColor, darken } from "@operational/utils"

export interface IProps {
  id?: string | number
  style?: {}
  className?: string
  children: React.ReactNode
  onClick?: any
  theme?: Theme
}

const style = ({ theme, color }: { theme: Theme; color?: string }): {} => {
  const backgroundColor = color ? hexOrColor(color)(theme.colors[color]) : theme.colors.gray90

  return {
    backgroundColor,
    position: "relative",
    zIndex: theme.baseZIndex + 1,
    margin: `0 ${theme.spacing * -0.5}px`,
    padding: `${theme.spacing}px`,
    minWidth: 200,
    borderRadius: 2,
    transition: ".1s background-color ease",
    color: readableTextColor(backgroundColor)(["black", "white"]),

    "& + &": {
      borderTop: `1px solid ${theme.colors.gray90}`
    },

    ":hover": {
      backgroundColor: darken(backgroundColor)(10)
    },

    ":focus": {
      outline: 0,
      backgroundColor: darken(backgroundColor)(15)
    },

    ":first-child": {
      marginTop: `${theme.spacing * -0.5}px`
    },

    ":last-child": {
      marginBottom: `${theme.spacing * -0.5}px`
    }
  }
}

const SideNavigationLink = (props: IProps) => (
  <div
    key={props.id}
    style={props.style}
    className={props.className}
    onClick={props.onClick}
    role="button"
    tabIndex={-1}
  >
    {props.children}
  </div>
)

export default glamorous(SideNavigationLink)(style)
export { SideNavigationLink, style } // for testing.
