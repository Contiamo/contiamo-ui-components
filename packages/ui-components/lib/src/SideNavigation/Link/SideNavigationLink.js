"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var contiamo_ui_utils_1 = require("contiamo-ui-utils");
var style = function (_a) {
    var theme = _a.theme, color = _a.color;
    var backgroundColor = color ? contiamo_ui_utils_1.hexOrColor(color)(theme.colors[color]) : theme.greys["100"];
    return {
        position: "relative",
        zIndex: (theme.baseZIndex || 0) + 1,
        margin: "0 " + theme.spacing * -0.5 + "px",
        padding: theme.spacing + "px",
        minWidth: 200,
        borderRadius: 2,
        transition: ".1s background-color ease",
        backgroundColor: backgroundColor,
        color: contiamo_ui_utils_1.readableTextColor(backgroundColor)(["black", "white"]),
        "& + &": {
            borderTop: "1px solid " + theme.greys["100"]
        },
        ":hover": {
            backgroundColor: contiamo_ui_utils_1.darken(backgroundColor)(10)
        },
        ":first-child": {
            marginTop: theme.spacing * -0.5 + "px"
        },
        ":last-child": {
            marginBottom: theme.spacing * -0.5 + "px"
        }
    };
};
exports.style = style;
var SideNavigationLink = function (_a) {
    var className = _a.className, children = _a.children, onClick = _a.onClick;
    return (React.createElement("div", { className: className, onClick: onClick, role: "button", tabIndex: -1 }, children));
};
exports.SideNavigationLink = SideNavigationLink;
exports.default = glamorous_1.default(SideNavigationLink)(style);
//# sourceMappingURL=SideNavigationLink.js.map