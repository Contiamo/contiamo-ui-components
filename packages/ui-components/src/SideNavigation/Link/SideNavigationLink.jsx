// @flow
import React from "react"
import type { Node } from "react"
import glamorous from "glamorous"

import { hexOrColor, readableTextColor, darken } from "contiamo-ui-utils"

type props = {
  className: string,
  children: Node,
  onClick?: any,
  theme?: THEME,
}

const SideNavigationLink = ({ className, children, onClick }: props): React$Element<*> =>
    <div className={className} onClick={onClick} role="button" tabIndex="-1">
      {children}
    </div>,
  style = ({ theme, color }: { theme: THEME, color?: string }): {} => {
    const backgroundColor = color ? hexOrColor(color)(theme.colors[color]) : theme.greys && theme.greys["100"]

    return {
      position: "relative",
      zIndex: (theme.baseZIndex || 0) + 1,
      margin: `0 ${theme.spacing * -0.5}px`,
      padding: `${theme.spacing}px`,
      minWidth: 200,
      borderRadius: 2,
      transition: ".1s background-color ease",
      backgroundColor,
      color: readableTextColor(backgroundColor)(["black", "white"]),

      "& + &": {
        borderTop: `1px solid ${theme.greys && theme.greys["100"]}`
      },

      ":hover": {
        backgroundColor: darken(backgroundColor)(10)
      },

      ":first-child": {
        marginTop: `${theme.spacing * -0.5}px`
      },

      ":last-child": {
        marginBottom: `${theme.spacing * -0.5}px`
      }
    }
  }

export default glamorous(SideNavigationLink)(style)
export { SideNavigationLink, style } // for testing.
