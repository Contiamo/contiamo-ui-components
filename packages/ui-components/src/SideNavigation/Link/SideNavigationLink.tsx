import * as React from "react"

import glamorous from "glamorous"

import { hexOrColor, readableTextColor, darken } from "contiamo-ui-utils"

type Props = {
  className: string
  children: Node
  onClick?: any
  theme?: Theme
}

const style = ({ theme, color }: { theme: Theme; color?: string }): {} => {
  const backgroundColor = color ? hexOrColor(color)(theme.colors.palette[color]) : theme.colors.palette.grey90

  return {
    backgroundColor,
    position: "relative",
    zIndex: theme.baseZIndex + 1,
    margin: `0 ${theme.spacing * -0.5}px`,
    padding: `${theme.spacing}px`,
    minWidth: 200,
    borderRadius: 2,
    transition: ".1s background-color ease",
    color: readableTextColor(backgroundColor)(["black", "white"]),

    "& + &": {
      borderTop: `1px solid ${theme.colors.palette.grey90}`
    },

    ":hover": {
      backgroundColor: darken(backgroundColor)(10)
    },

    ":focus": {
      outline: 0,
      backgroundColor: darken(backgroundColor)(15)
    },

    ":first-child": {
      marginTop: `${theme.spacing * -0.5}px`
    },

    ":last-child": {
      marginBottom: `${theme.spacing * -0.5}px`
    }
  }
}

const SideNavigationLink: React.SFC<Props> = ({ className, children, onClick }: Props) => (
  <div className={className} onClick={onClick} role="button" tabIndex={-1}>
    {children}
  </div>
)

export default glamorous(SideNavigationLink)(style)
export { SideNavigationLink, style } // for testing.
