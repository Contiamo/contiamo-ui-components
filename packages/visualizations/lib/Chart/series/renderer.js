"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var area_1 = require("./renderers/area");
var bars_1 = require("./renderers/bars");
var flag_1 = require("./renderers/flag");
var line_1 = require("./renderers/line");
var symbol_1 = require("./renderers/symbol");
var text_1 = require("./renderers/text");
var Renderer = /** @class */ (function () {
    function Renderer(state, events, el, data, options, series) {
        switch (options.type) {
            case "area":
                return new area_1.default(state, events, el.select("g.series-area"), data, options, series);
            case "bars":
                return new bars_1.default(state, events, el.select("g.series-bars"), data, options, series);
            case "flag":
                return new flag_1.default(state, events, el.select("g.series-flag"), data, options, series);
            case "line":
                return new line_1.default(state, events, el.select("g.series-line"), data, options, series);
            case "symbol":
                return new symbol_1.default(state, events, el.select("g.series-symbol"), data, options, series);
            case "text":
                return new text_1.default(state, events, el.select("g.series-text"), data, options, series);
        }
    }
    return Renderer;
}());
exports.default = Renderer;
//# sourceMappingURL=renderer.js.map