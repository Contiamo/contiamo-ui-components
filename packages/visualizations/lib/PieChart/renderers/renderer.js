"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var donut_1 = require("./donut");
var polar_1 = require("./polar");
var gauge_1 = require("./gauge");
// Factory Class
var Renderer = /** @class */ (function () {
    function Renderer(state, events, el, options) {
        switch (options.type) {
            case "donut":
                return new donut_1.default(state, events, el, options);
            case "polar":
                return new polar_1.default(state, events, el, options);
            case "gauge":
                return new gauge_1.default(state, events, el, options);
            default:
                throw new Error("invalid render type '" + options.type + "' specified");
        }
    }
    return Renderer;
}());
exports.default = Renderer;
//# sourceMappingURL=renderer.js.map