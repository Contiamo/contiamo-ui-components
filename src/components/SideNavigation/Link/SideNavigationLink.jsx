// @flow
import React from 'react';
import glamorous from 'glamorous';

import { hexOrColor, readableTextColor, darken } from '../../../utils/color';

type props = {
  className: string,
  children: mixed,
  onClick?: void,
  theme: THEME,
};

const SideNavigationLink = ({ className, children, onClick }: props): React$Element<*> =>
  (<div className={className} onClick={onClick}>
    {children}
  </div>);

const style = ({ theme, color }: { theme: THEME, color?: string }): {} => {
  const backgroundColor = color
    ? hexOrColor(color)(theme.colors[color])
    : theme.greys && theme.greys['100'];

  return {
    position: 'relative',
    zIndex: 2,
    margin: `0 ${theme.spacing * -0.5}px`,
    padding: `${theme.spacing}px`,
    minWidth: 200,
    borderRadius: 2,
    transition: '.1s background-color ease',
    backgroundColor,
    color: readableTextColor(backgroundColor)(['black', 'white']),

    '& + &': {
      borderTop: `1px solid ${theme.greys && theme.greys['100']}`,
    },

    ':hover': {
      backgroundColor: darken(backgroundColor)(10),
    },

    ':first-child': {
      marginTop: `${theme.spacing * -0.5}px`,
    },

    ':last-child': {
      marginBottom: `${theme.spacing * -0.5}px`,
    },
  };
};

export default glamorous(SideNavigationLink)(style);
