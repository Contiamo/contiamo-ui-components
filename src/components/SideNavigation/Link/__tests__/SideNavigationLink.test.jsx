// @flow
import React from 'react';
import renderer from 'react-test-renderer';

import SideNavigationLink from '../SideNavigationLink';

test('SideNavigationLink component renders', () => {
  const output: Object = renderer.create(<SideNavigationLink>Hi, I'm a Link</SideNavigationLink>);
  const tree: {} = output.toJSON();
  expect(tree).toMatchSnapshot();
});
