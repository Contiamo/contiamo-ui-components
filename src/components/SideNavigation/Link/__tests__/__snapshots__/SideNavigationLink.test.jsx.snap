// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigationLink Should render properly 1`] = `
<div
  className="hi"
>
  Hi, I'm a Link
</div>
`;
